src/test/java/com/wikia/webdriver/testcases/adstests/TestAdsIncontentLeaderboardOasis.java

```diff
@@ -29,6 +29,8 @@ public void adsIncontentLeaderboard(String wikiName,
     new AdsBaseObject(driver, url)
         .triggerIncontentLeaderboard()
         .verifyLineItemId(AdsContent.INCONTENT_LEADERBOARD, lineItemId)
-        .verifyIframeSize(AdsContent.INCONTENT_LEADERBOARD, "gpt", slotWidth, slotHeight);
+        .verifyIframeSize(AdsContent.INCONTENT_LEADERBOARD, "gpt", slotWidth, slotHeight)
+        .waitForJavaScriptTruthy(
+            "window.getComputedStyle(document.querySelector('#INCONTENT_LEADERBOARD'),':before').getPropertyValue('content') == '\"advertisement\"'");
```

Comment:
In this case, could you create a separate generic method for getting information from CSS pseudo-elements which we can re-use in the future?